docs/source/reference/utils.rst

```diff
@@ -3,6 +3,23 @@
 Utilities
 =========
 
+Printing
+========
+
+
+.. function:: numba.misc.utils.printf(format, ...)
+
+   Writes the C string pointed by `format` to the standart output (stdout).
```

Comment:
as a user, the reference to "C string" makes me wonder if I need to do anything special before being able to use this function. I would add two examples (re-using the test code) to show how to pass an integer and a string to `printf`. I would also explain the difference between using `print` and `printf`? does the first one use the python machinery, and the second not? Is `printf` faster, more robust? When should it be used instead of `print`?

Reply:
I can provide a few answers here:

> if I need to do anything special before being able to use this function

No

> I would also explain the difference between using print and printf? does the first one use the python machinery, and the second not?

I agree. Yes. The `printf` used here is a C implementation of https://en.wikipedia.org/wiki/Printf_format_string

> Is printf faster, more robust?

It is faster for sure. It is as robust as is C `printf`. Since it only reads memory, it should be hard to crash the program: when the program crashes on the `printf` call, it is usually the argument expressions that cause the crash, not `printf` function.

> When should it be used instead of print?

Never :), `printf` is not a replacement of `print`. `printf` is useful for debugging LLVM IR as it allows printing the actual memory values of low-level variables.

Reply:
thanks @pearu . @guilhermeleobas I think the above is worth writing the documentation.